File: services/web/src/types/session.ts
export interface Session {
  id: string;
  user_id: string;
  duration_ms?: number;
  event_count?: number;
  error_count?: number;
  video_path?: string;
  status: string;
  created_at: string;
  completed_at?: string;
  metadata?: Record<string, unknown>;
}

export interface SessionEvent {
  id?: string;
  event_type: string;
  timestamp_ms: number;
  x: number;
  y: number;
  target: string;
  payload: string | Record<string, unknown> | null;
}

export interface SessionDetail {
  session: Session;
  events: SessionEvent[];
}

File: services/ingestion/internal/models/session.go
package models

import (
	"time"
)

// Session represents a recorded user session.
type Session struct {
	ID          string     `json:"id"`
	ProjectID   string     `json:"project_id"`
	UserID      string     `json:"user_id"`
	DurationMs  *int64     `json:"duration_ms"`
	VideoPath   *string    `json:"video_path"`
	EventCount  int        `json:"event_count"`
	ErrorCount  int        `json:"error_count"`
	Metadata    *string    `json:"metadata"`
	Status      string     `json:"status"`
	CreatedAt   time.Time  `json:"created_at"`
	CompletedAt *time.Time `json:"completed_at"`
}
